src/components/Courses/Courses.jsx

```diff
@@ -0,0 +1,54 @@
+import EmptyCourseList from '../EmptyCourseList/EmptyCourseList';
+import CourseCard from './components/CourseCard/CourseCard';
+import SearchBar from './components/SearchBar/SearchBar';
+
+import styles from '../Courses/Courses.module.css';
+
+import { useState } from 'react';
+
+const Courses = ({ course, author, ShowCourseInfo, isAuthenticated }) => {
+	const [data, setData] = useState(null);
+	const [filteredCourses, setFilteredCourses] = useState(null);
+
+	const getData = (input) => {
+		setData(input.target.value);
+	};
+
+	const buttonHandler = () => {
+		if (data === null) return course;
+		const lowercasedData = data.toLowerCase();
+		const coursefiltered = course.filter((el) =>
+			[
+				'id',
+				'title',
+				'description',
+				'duration',
+				'authors',
+				'creationDate',
+			].some((item) => String(el[item]).toLowerCase().includes(lowercasedData))
+		);
+		setFilteredCourses(coursefiltered);
+	};
+
+	if (course) {
+		return (
+			<div className={styles.container}>
+				<div className={styles.courseCard}>
+					<div className={styles.searchBar}>
+						<SearchBar getData={getData} buttonHandler={buttonHandler} />
```

Comment:
Please use names that describe what your function does. 
Example: getDataFromTheLocalStorage, extractDataFromTheStore, calculateAveragePostReactions

```suggestion
						<SearchBar onSearchChange={onSearchChange} handleSearch={handleSearch} />
```